src/Route.php

```diff
@@ -4,98 +4,100 @@
 
 namespace Yiisoft\Router;
 
+use Attribute;
 use InvalidArgumentException;
 use RuntimeException;
 use Stringable;
 use Yiisoft\Http\Method;
-use Yiisoft\Middleware\Dispatcher\MiddlewareDispatcher;
 
 use function in_array;
 
 /**
  * Route defines a mapping from URL to callback / name and vice versa.
  */
+#[Attribute(Attribute::TARGET_METHOD | Attribute::TARGET_CLASS | Attribute::IS_REPEATABLE)]
 final class Route implements Stringable
 {
-    private ?string $name = null;
+    private const POSSIBLE_METHODS = [
+        Method::GET,
+        Method::POST,
+        Method::PUT,
+        Method::DELETE,
+        Method::PATCH,
+        Method::HEAD,
+        Method::OPTIONS,
+    ];
 
-    /**
-     * @var string[]
-     */
-    private array $hosts = [];
-    private bool $override = false;
     private bool $actionAdded = false;
-
-    /**
-     * @var array[]|callable[]|string[]
-     */
-    private array $middlewareDefinitions = [];
-
-    private array $disabledMiddlewareDefinitions = [];
-
-    /**
-     * @var array<string,string>
-     */
-    private array $defaults = [];
+    private array $builtMiddlewares = [];
 
     /**
      * @param string[] $methods
+     * @param string[] $hosts
+     * @param array[]|callable[]|string[] $middlewares
+     * @param array<string,string> $defaults
      */
-    private function __construct(
+    public function __construct(
         private array $methods,
         private string $pattern,
-        private ?MiddlewareDispatcher $dispatcher = null
+        private ?string $name = null,
+        private array $middlewares = [],
+        /**
+         * Excludes middleware from being invoked when action is handled.
```

Comment:
phpdoc doesn't work like that for properties of constructor. Should be moved to `__construct()` block.